packages/thycotic_ss/data_stream/logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,495 @@
+        "IMPORTSECRETS - IMPORTED":                     { "category": ["iam"], "type": ["change"] }
+        "UNLIMITEDADMIN - ENABLE":                      { "category": ["iam"], "type": ["change"] }
+        "UNLIMITEDADMIN - DISABLE":                     { "category": ["iam"], "type": ["change"] }
+        "USERAUDIT - EXPIRENOW":                        { "category": ["iam"], "type": ["info"] }
+        "LICENSES - EXPIRES30DAYS":                     { "category": ["configuration"], "type": ["info"] }
+        "POWERSHELLSCRIPT - CREATE":                    { "category": ["iam"], "type": ["creation"] }
+        "POWERSHELLSCRIPT - DEACTIVATE":                { "category": ["iam"], "type": ["change"] }
+        "POWERSHELLSCRIPT - EDIT":                      { "category": ["iam"], "type": ["change"] }
+        "POWERSHELLSCRIPT - REACTIVATE":                { "category": ["iam"], "type": ["change"] }
+        "POWERSHELLSCRIPT - VIEW":                      { "category": ["iam"], "type": ["access"] }
+      source: >
+        def schemaId = ctx.cef.name.toString();
+        def schema = params[schemaId];
+        if (schema != null) {
+          if (ctx.event == null) {
+            ctx.event = new HashMap();
+          }
+          ctx.event.category = schema.category;
+          ctx.event.type = schema.type;
+        }
+
+  - grok:
+      field: cef.name
+      patterns:
+        - '%{DATA:event.provider} - %{GREEDYDATA:event.action}'
+        - '%{DATA:event.provider} %{GREEDYDATA:event.action}'
+        - '%{GREEDYDATA:event.action}'
+      ignore_missing: true
+      description: "Parse cef.name into more usable ECS event.provider and event.action fields"
+
+  - lowercase:
+      field: event.provider
+      ignore_missing: true
+
+  - lowercase:
+      field: event.action
+      ignore_missing: true
+
+  - convert:
+      field: cef.device.event_class_id
+      type: string
+      target_field: event.code
+      ignore_missing: true
+      description: "The CEF name is the basic event description used by Thycotic SS"
+
+  - grok:
+      field: _tmp.pre_cef
+      patterns:
+        - '%{TIME} %{HOSTNAME:log.syslog.hostname}'
+      ignore_missing: true
+      if: ctx._tmp?.pre_cef != null && ctx._tmp?.pre_cef != ""
+      description: "If Elastic Agent read the event from a log file, try to extract the hostname of the original sender from before the CEF:0 header"
+
+  - convert:
+      field: log.syslog.hostname
+      type: string
+      target_field: host.name
+      ignore_missing: true
+      description: "Store syslog source hostname if available to host.name"
+
+  - append:
+      field: related.hosts
+      value:
+        - '{{host.name}}'
+      if: ctx.host?.name != null && ctx.host?.name != ""
+
+  - grok:
+      field: log.source.address
+      patterns:
+        - '%{IP:_tmp.host.ip}:%{NUMBER}'
+        - '%{IP:_tmp.host.ip}'
+      ignore_missing: true
+      description: "Parse syslog source address if available to host.ip"
```

```suggestion
  - grok:
      field: log.source.address
      patterns:
        - '%{IP:_tmp.host.ip:ip}:%{NUMBER}'
        - '%{IP:_tmp.host.ip:ip}'
      ignore_missing: true
      on_failure:
        - remove:
            field: _tmp.host.ip
      description: "Parse syslog source address if available to host.ip"
```
This makes sure the type is IP and that if the text is incorrect we don't fail out later. Similar below.